### PR TITLE
Possible fix for dependabot issues and add CodeQL workflow

### PR DESCRIPTION
**Pull request by hjpotter92**
**What does this pull request do? Explain your changes. (required)**

- The dependabot generated PRs don't have access to secrets. This PR tries to resolve that.
- Adds CodeQL workflow to be run parallel with tests.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,13 +1,18 @@
 name: Build binaries
 
 on:
+  pull_request_target:
```
Comment (leszko) [inline .github/workflows/build.yaml:4]: Isn't this a security hole? E.g., a malicious user could write a code that steals our secrets and send a PR.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
   pull_request:
   push:
     branches:
       - master
     tags:
       - "v*"
 
+concurrency:
```
Comment (leszko) [inline .github/workflows/build.yaml:12]: Is it related to `The dependabot generated PRs don't have access to secrets`? I can't connect the dots how it's related.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -93,6 +94,42 @@ jobs:
         uses: codecov/codecov-action@v3
         with:
           files: ./cover.out
-          name: go-livepeer
+          name: ${{ github.event.repository.name }}
           verbose: true
           gcov_ignore: ./eth/stubclient.go
+
+  codeql:
```
Comment (leszko) [inline .github/workflows/test.yaml:101]: Is this related to `The dependabot generated PRs don't have access to secrets`?

**Review by leszko — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (hjpotter92) [reply .github/workflows/build.yaml:4]: dependabot secrets are stored separate from the default ones: 

image

but any PR (from external users) would still need to be approved by repo owner before the CI picks up the job. The same is true for `pull_request` based trigger.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (hjpotter92) [reply .github/workflows/build.yaml:12]: concurrency avoids running the same pipeline multiple times; and with dependabot generating/rebasing the PRs, it is just a good practice to use concurrency.

File: .github/workflows/test.yaml (hunk 3 above)
Comment (hjpotter92) [reply .github/workflows/test.yaml:101]: codeql itself isn't related to dependabot.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (leszko) [reply .github/workflows/build.yaml:4]: Ok, then it makes sense.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (leszko) [reply .github/workflows/build.yaml:12]: 👍

File: .github/workflows/test.yaml (hunk 3 above)
Comment (leszko) [reply .github/workflows/test.yaml:101]: ok, so could you please send it as a separate PR or update the description of this PR?

Comment (hjpotter92) [reply .github/workflows/test.yaml:101]: updated the description here.

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -93,6 +94,42 @@ jobs:
...
+        uses: github/codeql-action/autobuild@v1
+
+      # ℹ️ Command-line programs to run using the OS shell.
+      # 📚 https://git.io/JvXDl
+
+      # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+      #    and modify them (or add more) to build your code if your project
+      #    uses a compiled language
+
+      #- run: |
```
Comment (leszko) [inline .github/workflows/test.yaml]: Could you maybe clean up this commented-out code?

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -93,6 +94,42 @@ jobs:
...
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: go
+
+      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below)
```
Comment (leszko) [inline .github/workflows/test.yaml]: I'm not sure I understand this command. When should we "remove it"?

**Review by leszko — APPROVED** [review]
LGTM after cleaning up comments.

**Pull request merged**